pkg/cmd/project/link/link.go

```diff
@@ -0,0 +1,162 @@
+			client, err := client.New(f)
+			if err != nil {
+				return err
+			}
+
+			if len(args) == 1 {
+				num, err := strconv.ParseInt(args[0], 10, 32)
+				if err != nil {
+					return cmdutil.FlagErrorf("invalid number: %v", args[0])
+				}
+				opts.number = int32(num)
+			}
+
+			config := linkConfig{
+				httpClient: f.HttpClient,
+				client:     client,
+				opts:       opts,
+				io:         f.IOStreams,
+			}
+
+			if err := cmdutil.MutuallyExclusive("specify only one of `--repo` or `--team`", opts.repo != "", opts.team != ""); err != nil {
+				return err
+			}
+			if err := cmdutil.MutuallyExclusive("specify either `--repo` or `--team`", opts.repo == "", opts.team == ""); err != nil {
+				return err
+			}
+
+			// allow testing of the command without actually running it
+			if runF != nil {
+				return runF(config)
+			}
+			return runLink(config)
+		},
+	}
+
+	linkCmd.Flags().StringVar(&opts.owner, "owner", "", "Login of the owner. Use \"@me\" for the current user.")
+	linkCmd.Flags().StringVarP(&opts.repo, "repo", "R", "", "The repository to be linked to this project")
+	linkCmd.Flags().StringVarP(&opts.team, "team", "T", "", "The team to be linked to this project")
+	cmdutil.AddFormatFlags(linkCmd, &opts.exporter)
+
+	return linkCmd
+}
+
+func runLink(config linkConfig) error {
+	canPrompt := config.io.CanPrompt()
+	owner, err := config.client.NewOwner(canPrompt, config.opts.owner)
+	if err != nil {
+		return err
+	}
+
+	project, err := config.client.NewProject(canPrompt, owner, config.opts.number, false)
+	if err != nil {
+		return err
+	}
+	config.opts.projectID = project.ID
+
+	httpClient, err := config.httpClient()
+	if err != nil {
+		return err
+	}
+	c := api.NewClientFromHTTP(httpClient)
+
+	if config.opts.repo != "" {
+		return linkRepo(c, owner, config)
+	} else if config.opts.team != "" {
+		return linkTeam(c, owner, config)
+	}
+	return nil
+}
+
+func linkRepo(c *api.Client, owner *queries.Owner, config linkConfig) error {
+	repo, err := api.GitHubRepo(c, ghrepo.New(owner.Login, config.opts.repo))
+	if err != nil {
+		return err
+	}
+	config.opts.repoID = repo.ID
+
+	result, err := config.client.LinkProjectToRepository(config.opts.projectID, config.opts.repoID)
+	if err != nil {
+		return err
+	}
+
+	if config.opts.exporter != nil {
+		return config.opts.exporter.Write(config.io, result)
+	}
+	return printResults(config, result.URL)
+}
+
+func linkTeam(c *api.Client, owner *queries.Owner, config linkConfig) error {
+	team, err := api.OrganizationTeam(c, ghrepo.New(owner.Login, ""), config.opts.team)
+	if err != nil {
+		return err
+	}
+	config.opts.teamID = team.ID
+
+	result, err := config.client.LinkProjectToTeam(config.opts.projectID, config.opts.teamID)
+	if err != nil {
+		return err
+	}
+
+	if config.opts.exporter != nil {
+		return config.opts.exporter.Write(config.io, result)
+	}
+	return printResults(config, result.URL)
+}
+
+func printResults(config linkConfig, url string) error {
```

Comment:
Is there something that you wanted to get out of the `Team` or `Repository` URL, or wanted to have scripted with the exporter? I'm not sure what the workflow is that would make this useful information to have and I'm always in favour of removing code unless we have a specific need.

Reply:
Just like other `gh projects` commands, print out the URL upon successful execution. Should we refrain from printing anything after a successful execution?

Reply:
I'm not sure. I think if we were to print something it should be like `"Linked X to Y". What do you think?

Reply:
It's good. It looks intuitive.